### ISSUE TITLE
sockmap: failed to pin map and attach program

### ISSUE THREAD

**Issue by vadorovsky** [cilium/cilium#6972]
```
level=info msg="     _ _ _" subsys=daemon                                                                                                                                                                                                   
level=info msg=" ___|_| |_|_ _ _____" subsys=daemon                                                                                                                                                                                         
level=info msg="|  _| | | | | |     |" subsys=daemon                                      
level=info msg="|___|_|_|_|___|_|_|_|" subsys=daemon                                  
level=info msg="Cilium 1.4.90 d1db71e99 2019-02-06T13:27:17+01:00 go version go1.10.8 linux/amd64" subsys=daemon                                                                                                                            
level=info msg="cilium-envoy  version: 84ee839e1d78ef858a39e390288ad417d35bb1d4/1.9.0-dev/Modified/RELEASE" subsys=daemon                                                                                                                    level=info msg="clang (7.0.0) and kernel (4.20.2) versions: OK!" subsys=daemon                                                                                                                                                               level=info msg="linking environment: OK!" subsys=daemon     
level=info msg="bpf_requirements check: OK!" subsys=daemon                                                 
level=info msg="bpf_features check: OK!" subsys=daemon                                                                                                                                                                                       level=info msg="Mounted BPF filesystem /sys/fs/bpf" subsys=bpf               
level=info msg="Waiting for consul to elect a leader" subsys=kvstore                                                                                                                                                                         level=info msg="Valid label prefix configuration:" subsys=labels-filter
level=info msg=" - :io.kubernetes.pod.namespace" subsys=labels-filter                                                                                                                                                                        level=info msg=" - :io.cilium.k8s.namespace.labels" subsys=labels-filter
level=info msg=" - !:io.kubernetes" subsys=labels-filter                                                                                                                                                                                     level=info msg=" - !:.*kubernetes.io" subsys=labels-filter                         
level=info msg=" - !:pod-template-generation" subsys=labels-filter                                                                                                                                                                           level=info msg=" - !:pod-template-hash" subsys=labels-filter
level=info msg=" - !:controller-revision-hash" subsys=labels-filter                                                                                                                                                                          level=info msg=" - !:annotation.cilium.io/" subsys=labels-filter
level=info msg=" - !:annotation.cilium-identity" subsys=labels-filter                                                                                                                                                                        level=info msg=" - !:annotation.sidecar.istio.io" subsys=labels-filter                                    
level=info msg=" - !:annotation.etcd.version" subsys=labels-filter                                                                                                                                                                           level=info msg=" - !:etcd_node" subsys=labels-filter                      
level=info msg="Initializing daemon" subsys=daemon                                                                                                                                                                                           level=info msg="Detected MTU 1500" subsys=mtu                                                                       
level=info msg="Clearing leftover Cilium veths" subsys=daemon                                                                                                                                                                                level=info msg="Initializing node addressing" subsys=daemon                                                   
level=info msg="Restored IPv4 internal node IP: 10.6.0.1" subsys=node                                                                                                                                                                        level=info msg="Restored IPv6 router IP: f00d::a06:0:0:1" subsys=node                                             
level=info msg="Using autogenerated IPv4 allocation range" subsys=node v4Prefix=10.6.0.0/16                                                                                                                                                  level=info msg="Using autogenerated IPv6 allocation range" subsys=node v6Prefix="f00d::a06:0:0:0/96"           
level=info msg="Initializing IPAM" subsys=daemon
level=info msg="Container runtime options set: endpoint=/var/run/containerd/containerd.sock,endpoint=/var/run/crio.sock,datapath-mode=veth,endpoint=unix:///var/run/docker.sock" subsys=daemon  
level=info msg="Restoring endpoints from former life..." subsys=daemon                                                                                                                                                                       level=info msg="Endpoints restored" count.restored=0 count.total=1 subsys=daemon
level=info msg="Addressing information:" subsys=daemon                                                                                                                                                                                       level=info msg="  Cluster-Name: default" subsys=daemon                                                         
level=info msg="  Cluster-ID: 0" subsys=daemon                                                                                                                                                                                               level=info msg="  Local node-name: mrostecki-suse" subsys=daemon
level=info msg="  Node-IPv6: 2a02:2450:1010:857:21ce:dc77:83e1:c450" subsys=daemon                                                                                                                                                           level=info msg="  IPv6 node prefix: f00d::a06:0:0:0/96" subsys=daemon
level=info msg="  IPv6 allocation prefix: f00d::a06:0:0:0/112" subsys=daemon                                                                                                                                                                 level=info msg="  IPv6 router address: f00d::a06:0:0:1" subsys=daemon                                                     
level=info msg="  External-Node IPv4: 192.168.0.6" subsys=daemon                                                                                                                                                                             level=info msg="  Internal-Node IPv4: 10.6.0.1" subsys=daemon                            
level=info msg="  Cluster IPv4 prefix: 10.0.0.0/8" subsys=daemon                                                                                                                                                                             level=info msg="  IPv4 allocation prefix: 10.6.0.0/16" subsys=daemon           
level=info msg="  Loopback IPv4: 10.6.111.42" subsys=daemon                                                                                                                                                                                  level=info msg="Adding local node to cluster" subsys=daemon                                                                                      
level=info msg="Initializing identity allocator" subsys=identity-cache                                                                                                                                                                       level=info msg="Starting to watch allocation changes" kvstoreErr="<nil>" kvstoreStatus="Consul: 10.88.0.10:8300" prefix=cilium/state/identities/v1/id subsys=allocator
level=info msg="Sockmap disabled." subsys=sockops                                                                                                                                                                                            level=info msg="Sockmsg Disabled." subsys=sockops                                                                                                                                                                                           
level=error msg="Failed to pin map 0(cilium_sock_ops_map): exit status 255" subsys=sockops                                                                                                                                                   level=error msg="Failed to attach prog(551) to map(0): exit status 255" subsys=sockops
```

**Comment by vadorovsky** [comment]
I'm running Cilium from master just on my host (with openSUSE Tumbleweed) and I built bpftool from bpf-next tree. Yeah, I know, that log says almost nothing. I will try to dig more into that issue myself.

**Comment by borkmann** [comment]
Heh, nice coincidence, fix is already out (#6988)
